cmake/CUDAConfig.cmake

```diff
@@ -0,0 +1,32 @@
+if(NOT ${CUDA_ARCH} STREQUAL "")
+	#SET(CUDA_SEPARABLE_COMPILATION ON) # for -dc; no need
+	find_package(CUDA)
+	if(CUDA_FOUND)
+		message("find CUDA!!!")
+	endif()
+	
+	enable_language(CUDA)	
+	set(CMAKE_CUDA_FLAGS "-O2 -Xptxas=-v -m64 -use_fast_math")
+	set(CMAKE_CXX_FLAGS "-DCUBLAS_USE_THUNKING -O2 -I${CUDA_INCLUDE_DIRS}")
+	
+	#message("CUDA_INCLUDE_DIRS is ${CUDA_INCLUDE_DIRS}")
+	#message("CUDA_LIBRARIES  is ${CUDA_LIBRARIES}")
+	#message("CUDA_TOOLKIT_ROOT_DIR is ${CUDA_TOOLKIT_ROOT_DIR}")	#this is CUDA_HOME
+	#message("CUDA_cublas_LIBRARY is ${CUDA_cublas_LIBRARY}")	
+	
+	include_directories(${CUDA_INCLUDE_DIRS})	
+
+	add_definitions(-DCUDA)
+	add_definitions(-DDEBUG)
+		
+	if(${CUDA_ARCH} STREQUAL "pascal")
```

Comment:
This block doesn't properly duplicate the logic from the configure script (see my implementation in CMakeLists.txt).

Reply:
I'm a bit confused here. `add_definitions` won't work in this case? What's the difference between `add_definitions`, `compile_options` and`target_compile_options`?

Reply:
What I meant is that the `if(${CUDA_ARCH} STREQUAL "pascal")` block doesn't duplicate the logic correctly.  You should also have code to auto-set CUDA_ARCH if the user doesn't provide it, like how the configure script enables different architectures based on CUDA version.

I already implemented this starting [here](https://github.com/multiplemonomials/QUICK/blob/080ab85894ecd3dbe689e7d804d24743142e6d98/CMakeLists.txt#L80), you should be able to just use my implementation (changing some of the variable names).

Reply:
This block has been fixed per your suggestion. I just have two naive questions:
1. What's the difference between `CUDA_NVCC_FLAGS` and `CMAKE_CUDA_FLAGS`?
2. For `CUDA_VERSION`~[9.0,10.0) you used `list(APPEND CUDA_NVCC_FLAGS ${SM60FLAGS} ${SM70FLAGS})`. Does that mean for instance `CUDA/9.1.85` would have two sets of `arch` and `code` values?

Reply:
CUDA_NVCC_FLAGS is the FindCUDA equivalent of CMAKE_CUDA_FLAGS.  Also, CMAKE_CUDA_FLAGS is space-seperated while CUDA_NVCC_FLAGS is a list.  You should be using CMAKE_CUDA_FLAGS since you're not using FindCUDA.  Also, for the same reason, you should use CMAKE_CUDA_COMPILER_VERSION instead of CUDA_VERSION.  The logic is a lot different between the new and old CUDA libraries.

Reply:
> Does that mean for instance CUDA/9.1.85 would have two sets of arch and code values?

Yep!  This usage causes nvcc to compile the file twice, once for each architecture.